test/Microsoft.TemplateEngine.Cli.UnitTests/ParserTests/TabCompletionTests.Approval.cs

```diff
@@ -0,0 +1,123 @@
+        }
+
+#pragma warning disable xUnit1004 // Test methods should not be skipped
+        [Fact (Skip = "https://github.com/dotnet/command-line-api/blob/main/src/System.CommandLine/Parsing/ParseResultExtensions.cs#L285-L289; the tab completion also contains the results from parent command")]
+#pragma warning restore xUnit1004 // Test methods should not be skipped
+        public Task Update_GetAllSuggestions()
+        {
+            ITemplateEngineHost host = TestHost.GetVirtualHost(additionalComponents: BuiltInTemplatePackagesProviderFactory.GetComponents(includeTestTemplates: false));
+            var myCommand = NewCommandFactory.Create("new", host, new TelemetryLogger(null, false), new NewCommandCallbacks());
+
+            var parseResult = myCommand.Parse("new update ");
+            var result = parseResult.GetSuggestions().ToArray();
+
+            return Verifier.Verify(result, _verifySettings.Settings);
+        }
+
+        [Fact]
+        public Task List_GetAllSuggestions()
+        {
+            ITemplateEngineHost host = TestHost.GetVirtualHost(additionalComponents: BuiltInTemplatePackagesProviderFactory.GetComponents(includeTestTemplates: false));
+            var myCommand = NewCommandFactory.Create("new", host, new TelemetryLogger(null, false), new NewCommandCallbacks());
+
+            var parseResult = myCommand.Parse("new list ");
+            var result = parseResult.GetSuggestions().ToArray();
+
+            return Verifier.Verify(result, _verifySettings.Settings);
+        }
+
+        [Fact]
+        public Task Search_GetAllSuggestions()
+        {
+            ITemplateEngineHost host = TestHost.GetVirtualHost(additionalComponents: BuiltInTemplatePackagesProviderFactory.GetComponents(includeTestTemplates: false));
+            var myCommand = NewCommandFactory.Create("new", host, new TelemetryLogger(null, false), new NewCommandCallbacks());
+
+            var parseResult = myCommand.Parse("new search ");
+            var result = parseResult.GetSuggestions().ToArray();
+
+            return Verifier.Verify(result, _verifySettings.Settings);
+        }
+
+        [Fact]
+        public Task Create_GetAllSuggestions()
+        {
+            ITemplateEngineHost host = TestHost.GetVirtualHost(additionalComponents: BuiltInTemplatePackagesProviderFactory.GetComponents(includeTestTemplates: false));
+            var myCommand = NewCommandFactory.Create("new", host, new TelemetryLogger(null, false), new NewCommandCallbacks());
+
+            var parseResult = myCommand.Parse("new create ");
+            var result = parseResult.GetSuggestions().ToArray();
+
+            return Verifier.Verify(result, _verifySettings.Settings);
+        }
+
+#pragma warning disable xUnit1004 // Test methods should not be skipped
+        [Fact(Skip = "https://github.com/dotnet/command-line-api/blob/main/src/System.CommandLine/Parsing/ParseResultExtensions.cs#L285-L289; the tab completion also contains the results from parent command")]
```

Comment:
this test doesn't work now, the result also contain the result from base command, due to https://github.com/dotnet/command-line-api/blob/main/src/System.CommandLine/Parsing/ParseResultExtensions.cs#L285-L289